
# HG changeset patch
# User Maxim Dounin <mdounin@mdounin.ru>
# Date 1368283770 -14400
# Node ID 9d83ec073c514acec7adc84ee15d74692a5def7a
# Parent  3494f14aa46a098bfcb4a1dc24acc7826cdd603b
Fixed build with --with-mail_ssl_module.

If nginx was compiled without --with-http_ssl_module, but with some
other module which uses OpenSSL (e.g. --with-mail_ssl_module), insufficient
preprocessor check resulted in build failure.  The problem was introduced
by e0a3714a36f8 (1.3.14).

Reported by Roman Arutyunyan.

diff -r 3494f14aa46a -r 9d83ec073c51 src/http/ngx_http.h
--- src/http/ngx_http.h	Sat May 11 18:49:19 2013 +0400
+++ src/http/ngx_http.h	Sat May 11 18:49:30 2013 +0400
@@ -89,7 +89,7 @@
 void ngx_http_init_connection(ngx_connection_t *c);
 void ngx_http_close_connection(ngx_connection_t *c);
 
-#ifdef SSL_CTRL_SET_TLSEXT_HOSTNAME
+#if (NGX_HTTP_SSL && defined SSL_CTRL_SET_TLSEXT_HOSTNAME)
 int ngx_http_ssl_servername(ngx_ssl_conn_t *ssl_conn, int *ad, void *arg);
 #endif
 
diff -r 3494f14aa46a -r 9d83ec073c51 src/http/ngx_http_request.c
--- src/http/ngx_http_request.c	Sat May 11 18:49:19 2013 +0400
+++ src/http/ngx_http_request.c	Sat May 11 18:49:30 2013 +0400
@@ -1955,7 +1955,7 @@
 
     hc = r->http_connection;
 
-#ifdef SSL_CTRL_SET_TLSEXT_HOSTNAME
+#if (NGX_HTTP_SSL && defined SSL_CTRL_SET_TLSEXT_HOSTNAME)
 
     if (hc->ssl_servername) {
         if (hc->ssl_servername->len == host->len
@@ -1986,7 +1986,7 @@
         return NGX_ERROR;
     }
 
-#ifdef SSL_CTRL_SET_TLSEXT_HOSTNAME
+#if (NGX_HTTP_SSL && defined SSL_CTRL_SET_TLSEXT_HOSTNAME)
 
     if (hc->ssl_servername) {
         ngx_http_ssl_srv_conf_t  *sscf;
@@ -2053,7 +2053,7 @@
 
         sn = virtual_names->regex;
 
-#ifdef SSL_CTRL_SET_TLSEXT_HOSTNAME
+#if (NGX_HTTP_SSL && defined SSL_CTRL_SET_TLSEXT_HOSTNAME)
 
         if (r == NULL) {
             ngx_http_connection_t  *hc;
@@ -2085,7 +2085,7 @@
             return NGX_DECLINED;
         }
 
-#endif /* SSL_CTRL_SET_TLSEXT_HOSTNAME */
+#endif /* NGX_HTTP_SSL && defined SSL_CTRL_SET_TLSEXT_HOSTNAME */
 
         for (i = 0; i < virtual_names->nregex; i++) {
 
diff -r 3494f14aa46a -r 9d83ec073c51 src/http/ngx_http_request.h
--- src/http/ngx_http_request.h	Sat May 11 18:49:19 2013 +0400
+++ src/http/ngx_http_request.h	Sat May 11 18:49:30 2013 +0400
@@ -295,7 +295,7 @@
     ngx_http_addr_conf_t             *addr_conf;
     ngx_http_conf_ctx_t              *conf_ctx;
 
-#ifdef SSL_CTRL_SET_TLSEXT_HOSTNAME
+#if (NGX_HTTP_SSL && defined SSL_CTRL_SET_TLSEXT_HOSTNAME)
     ngx_str_t                        *ssl_servername;
 #if (NGX_PCRE)
     ngx_http_regex_t                 *ssl_servername_regex;

